### ISSUE TITLE
Adding compilation / lex flags to a ninja running build

### ISSUE THREAD

**Issue by albert-github** [ninja-build/ninja#1436]
When building a project with make I can define a rule like:
    `/usr/bin/flex.exe $(LEX_FLAGS) -ofile.cpp file.l`
where LEX_FLAGS can be defined in the `Makefile` or outside the `Makefile` and can be altered by means of:
    `make LEX_FLAGS=-d`

I want to do the same with ninja, but I don't see a possibility to do it. I tried numerous possibilities but failed.

**Comment by jonesmz** [comment]
Would it be correct to say you are asking for ninja to support substituting a variable with the contents of an environment variable of the same name?

**Comment by albert-github** [comment]
Yes it is the behavior that the `make` and `nmake` command show and that would be very useful when `ninja` would also support this.

**Comment by jonesmz** [comment]
Thank you for the clarification!
